designer/report-designer-extension-pentaho/src/main/java/org/pentaho/reporting/designer/extensions/pentaho/repository/actions/ValidateLoginTask.java

```diff
@@ -91,4 +123,68 @@ public boolean validateLoginData() throws FileSystemException {
     }
     return true;
   }
+
+
+  /**
+   * A faster way to check the provided credentials by calling a servers tree API with a basic authentication.
+   *
+   * @return
+   */
+  public boolean validateLoginDataFast() {
+
+    //Backward compatibility
+    if ( loginData == null ) {
+      return true;
+    }
+
+    try {
+
+      final HttpClient httpClient = getHttpClient();
+      final URI uri = new URI( StringUtils.stripEnd( loginData.getUrl(), "/ " ) + TREE_API_URL_PART );
+      final Credentials credentials =
+        new UsernamePasswordCredentials( loginData.getUsername(), loginData.getPassword() );
+      final HttpState state = new HttpState();
+      state.setCredentials( new AuthScope( uri.getHost(), uri.getPort(), AuthScope.ANY_REALM ), credentials );
+      httpClient.setState( state );
+
+      //Remove trailing slashes and spaces
+      final HttpMethod get = new GetMethod( uri.toString() );
+      final String authenticationString = loginData.getUsername() + ":" + loginData.getPassword();
+      //Checked with a real Pentaho server - it supports UTF encoding for a basic authentication
+      final String base64Credentials =
+        new Base64Encoder().encode( authenticationString.getBytes( Charset.forName( "UTF-8" ) ) );
+      get.setRequestHeader( AUTHORIZATION, "Basic " + base64Credentials );
```

Comment:
You dont need that manual base64Credentials stuff. The HTTP-Client's Credentials above take care of the authentication for you.